platform/ios/app/MBXStateManager.h

```diff
@@ -0,0 +1,19 @@
+#import <Foundation/Foundation.h>
+#import "MBXViewController.h"
+@class MBXState;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MBXStateManager : NSObject
+
++ (instancetype) sharedManager;
+
+- (MBXState *)currentState;
+
+- (void)saveState:(MBXViewController*)mapViewController;
```

Comment:
In this case `MBXStateManager` main responsibility is save/update/reset `MBXState` objects. Passing a `MBXViewController` creates a dependency on the later, reducing the manager's re-usability. What if we want to use this manager in a different controller? we would be required to create an `MBXViewController` object just to use the manager.

Change this to:
`saveState:(MBXState *)state`